virtualmachine/vsphere/util.go

```diff
@@ -419,13 +419,23 @@ var reconfigureVM = func(vm *VM, vmMo *mo.VirtualMachine) error {
 		return err
 	}
 
+	dcMo, err := GetDatacenter(vm)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve datacenter: %s", err)
```

Comment:
use `return err`, same as rest of the code. Assuming caller is expecting/handling that.
You can still print the user friendly failure message before returning.